### PR TITLE
Update listing

### PR DESCRIPTION
**Pull request by ba1ciu**

### CODE REVIEW

File: chain/tests/e2e/marketplace/update_listing_test.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+	wasmcli "github.com/CosmWasm/wasmd/x/wasm/client/cli"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	clitestutil "github.com/cosmos/cosmos-sdk/testutil/cli"
+
+	"github.com/EmpowerPlastic/empowerchain/tests/e2e"
+	"github.com/EmpowerPlastic/empowerchain/x/plasticcredit/client/cli"
+)
+
+func (s *E2ETestSuite) TestUpdateListing() {
```
Comment (gjermundgaraba) [inline chain/tests/e2e/marketplace/update_listing_test.go:14]: I prefer to keep the e2e tests lighter and only test the things that can't be unit-tested directly in rust. These tests are very slow to run, after all.

Many of these tests seem redundant, given that they also exist in the unit tests, right?

File: cosmwasm/contracts/plastic-credit-marketplace/src/execute.rs [hunk 2]
```diff
@@ -28,17 +29,18 @@ pub fn execute_create_listing(
         return Err(ContractError::ZeroPrice {});
     }
 
-    let next_listing_id = NEXT_LISTING_ID.load(deps.storage)?;
+    if (LISTINGS.may_load(deps.storage, (info.sender.clone(), denom.clone()))?).is_some() {
```
Comment (gjermundgaraba) [inline cosmwasm/contracts/plastic-credit-marketplace/src/execute.rs:32]: Can you add a unit test that covers this scenario?

File: cosmwasm/contracts/plastic-credit-marketplace/src/execute.rs [hunk 3]
```diff
@@ -100,14 +102,78 @@ pub fn execute_buy_credits(
...
+    }
+
+    // Update the listing
+    listing.number_of_credits = number_of_credits;
+    listing.price_per_credit = price_per_credit.clone();
+    LISTINGS.save(deps.storage, (listing.owner.clone(), listing.denom.clone()), &listing)?;
+
+    let res = Response::new()
+        .add_attribute("action", "update_listing")
+        .add_attribute("listing_owner", info.sender)
```
Comment (gjermundgaraba) [inline cosmwasm/contracts/plastic-credit-marketplace/src/execute.rs:166]: Probably denom would be a good idea to add here too?

File: cosmwasm/contracts/plastic-credit-marketplace/src/query.rs [hunk 4]
```diff
@@ -8,29 +8,40 @@ pub const DEFAULT_LIMIT: u64 = 30;
 #[entry_point]
 pub fn query(deps: Deps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
     match msg {
+        QueryMsg::Listing {owner, denom} => to_binary(&listing(deps, owner, denom)?),
```
Comment (gjermundgaraba) [inline cosmwasm/contracts/plastic-credit-marketplace/src/query.rs:11]: Maybe at least a simple unit test for this one too (not found + found)

**Review by gjermundgaraba — CHANGES_REQUESTED** [review]

File: chain/tests/e2e/marketplace/update_listing_test.go (hunk 1 above)
Comment (ba1ciu) [reply chain/tests/e2e/marketplace/update_listing_test.go:14]: > I prefer to keep the e2e tests lighter and only test the things that can't be unit-tested directly in rust. These tests are very slow to run, after all.
> 
> Many of these tests seem redundant, given that they also exist in the unit tests, right?

Agreed. In that case I'd only keep 2 first tests as there are actual plastic credit transfers being made there. Makes sense? @gjermundgaraba

Comment (gjermundgaraba) [reply chain/tests/e2e/marketplace/update_listing_test.go:14]: Yes, that makes perfect sense :+1:

Comment (gjermundgaraba) [reply chain/tests/e2e/marketplace/update_listing_test.go:14]: Off-topic for this PR, but maybe we should do this for all our other e2e tests too. See which ones are actually redundant. They are getting _veeeery_ slow and heavy now.

Comment (ba1ciu) [reply chain/tests/e2e/marketplace/update_listing_test.go:14]: Agreed! Let's follow it [here](https://github.com/EmpowerPlastic/empowerchain/issues/559)!

**Review by gjermundgaraba — APPROVED** [review]

**Pull request merged**